static/js/emoji_picker.js

```diff
@@ -644,14 +644,33 @@ exports.register_click_handlers = function () {
     $(document).on('click', '.emoji-popover-emoji.composition', function (e) {
         var emoji_name = $(this).data("emoji-name");
         var emoji_text = ':' + emoji_name + ':';
-        compose_ui.insert_syntax_and_focus(emoji_text);
+        var message_edit_id = localStorage.getItem("message-id");
+        // Remove the message-id value from localStorage so that the selection of
+        // emoji in new message section works correclty.
+        localStorage.removeItem("message-id");
+        // The following check will return false if emoji was not selected in
+        // message edit form.
+        if (message_edit_id !== null) {
+            var message_edit_textarea = $("#message_edit_content_" + message_edit_id);
+            compose_ui.insert_syntax_and_focus(emoji_text, message_edit_textarea);
+        } else {
+            compose_ui.insert_syntax_and_focus(emoji_text);
+        }
         e.stopPropagation();
         emoji_picker.hide_emoji_popover();
     });
 
-    $("#compose").on("click", "#emoji_map", function (e) {
+    $("body").on("click", "#emoji_map", function (e) {
         e.preventDefault();
         e.stopPropagation();
+        // The data-message-id atribute is only present in the emoji icon present in
+        // the message edit form. So the following check will return false if this
+        // event was not fired from message edit form.
+        if ($(this).attr("data-message-id") !== undefined) {
+            // Store data-message-id value in localStorage so that its value can be
+            // further used to correclty find the message textarea element.
+            localStorage.setItem("message-id", $(this).attr("data-message-id"));
```

Comment:
Why are you using local storage, rather than just a global variable in this file?  There's no reason for this state to survive a browser reload.